geth/jail/jail_cell.go

```diff
@@ -51,20 +49,18 @@ func newJailCell(id string, vm *otto.Otto, lo *loop.Loop) (*JailCell, error) {
 	}, nil
 }
 
-// Copy returns a new JailCell instance with a new eventloop runtime associated with
-// the given cell.
-func (cell *JailCell) Copy() (common.JailCell, error) {
-	vmCopy := cell.vm.Copy()
-	return newJailCell(cell.id, vmCopy, loop.New(vmCopy))
-}
-
 // Fetch attempts to call the underline Fetch API added through the
 // ottoext package.
 func (cell *JailCell) Fetch(url string, callback func(otto.Value)) (otto.Value, error) {
+	cell.Lock()
 	if err := cell.vm.Set("__captureFetch", callback); err != nil {
+		defer cell.Unlock()
```

Comment:
I would rather leave it but moved right after `cell.Lock()` just because you won't forget to do `Unlock` in case you create another branch or something.